### ISSUE TITLE
No way to enable "no-extra-parens" on JSX and functions only

### ISSUE THREAD

**Issue by feross** [eslint/eslint#8174]
* **ESLint Version:** 3.16.1
* **Node Version:** 6.9.5
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```
"no-extra-parens": ["error", "all", {
  "conditionalAssign": false,
  "returnAssign": false,
  "nestedBinaryExpressions": false,
  "ignoreJSX": "multi-line"
 }]
```

**What did you do? Please include the actual source code causing the issue.**

```js
fn(x, (y ? 25 : 42))
```

**What did you expect to happen?**

No error.

**What actually happened? Please include the actual, raw output from ESLint.**

```
Gratuitous parentheses around expression. (no-extra-parens)
```

Backstory:

We were using the `no-extra-parens` rule with the `"functions"` option. But we wanted to also enforce that single-line JSX expressions don't have needless parens. It seemed that changing the rule from:

```
"no-extra-parens": ["error", "functions"]
```

To this:

```
"no-extra-parens": ["error", "all", {
  "conditionalAssign": false,
  "returnAssign": false,
  "nestedBinaryExpressions": false,
  "ignoreJSX": "multi-line"
 }]
```

Should have done the trick. But apparently `"all"` includes some checks for extra parens around expressions, but does not offer a way to disable those checks like with the other options (`conditionalAssign`, `returnAssign`, `nestedBinaryExpressions`, etc.)

We only want to prevent single-line JSX and functions from having extra parens, i.e. these two cases:

```js
const z = (<h1>hey</h1>)
((function foo() {}))()
```

Is it reasonable to allow users to disable these other checks with the second object parameter?

**Comment by vitorbal** [comment]
Hey @feross! Thanks for the issue :)

This seems reasonable, but I wonder how many new options would we need to disable checking all the cases you mention. Would something like a `"callExpressionArguments"` option suffice, or do you get errors for other types of expressions too?

**Comment by feross** [comment]
@vitorbal I'm not aware of any other cases that were causing warnings, but there might be some.

**Comment by vitorbal** [comment]
If something like a `"callExpressionArguments"` option is enough to satisfy the use-case for standard, I'd say we can treat this as an enhancement to add just that option.

We can worry about more options in the future, if the need ever comes.

**Comment by feross** [comment]
@vitorbal Sounds good to me!

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
